api/v1alpha3/tags.go

```diff
@@ -152,6 +158,18 @@ type BuildParams struct {
 	// Any additional tags to be added to the resource.
 	// +optional
 	Additional Tags
+
+	// machineName is the name of the Machine object associated with the resource, only applicable to vm resources
+	// When set, this will be applied as "MachineName" on AWS'
+	// +k8s:conversion-gen=false
+	machineName *string
```

Comment:
Yeah, I agree that we can move it to `AdditionalTags`. I'm wondering if we could still use the same api. Something like `BuildParams.WithMachineName(...)`, and `BuildParams.WithClouderProvider()` would be easier to read instead of having a function that sets all the tags?

Reply:
Good idea. I like the fluent nature of it.

Reply:
I've made the changes, PTAL.